samples/TEE_aware/gencreds/README.md

```diff
@@ -0,0 +1,228 @@
+# gencreds -  a TEE-aware program for Mystikos in C
+
+This sample guides users to create TEE-aware applications, which are
+essential to many confidential computing scenarios.
+
+It is recommended that you understand how to use EXT2 file system and how applications can be packaged
+Mystkos before proceeding with this sample. Please refer to the README for the helloworld sample
+to obtain the necessary information.
+
+
+## Set up pre-requisites
+
+It is essential for this program to be able to find Mystikos header files installed as part of the Mystikos package.
+To ensure that the Makefile can find the headers, export a variable `MYSTIKOS_INSTALL_DIR` and set it to the path of
+the Mystikos installation on the system. As an example, if Mystikos is installed on /opt/mystikos, then issue
+ the following command on the command prompt:
+
+```cmd
+
+export MYSTIKOS_INSTALL_DIR=/opt/mystikos.
+
+```
+## Functionality
+
+### The problem statement
```

Comment:
I prefer we keep the "problem statement and Mystikos solution" in a top level document because that's the unique strength of Mystikos that we want the users to be aware. If we put them here in a sample, the users would hardly notice.

Reply:
It is in doc as well. What do you suggest here?

Reply:
Then I suggest just providing a reference to the original doc to keep this readme short.

Reply:
For now, I have updated it to add ref to the original doc.
The original doc is outdated and references 'myst-gcc`. If you can approve this, we can then discuss what we should do about samples vs programming guides.